src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private Random random = new Random();
+    private int index = getRandom().nextInt(Colors.values().length);

```

Comment:
So there is no accident. When initializing a ColorSupplier object, the index is created once and will always be used only.